devito/function.py

```diff
@@ -348,14 +349,26 @@ def argument_values(self, alias=None, **kwargs):
         # Add value override for own data if it is provided
         if self.name in kwargs:
             new = kwargs.pop(self.name)
+            if isinstance(new, TimeFunction):
+                if new.save != self.save:
+                    raise TypeError("Operator generated for save=%s only," %
+                                    self.save +
+                                    " %s has to be created as " % new.name +
+                                    "TimeFunction(name=%s, grid=grid, save=%s)" %
+                                    (new.name, self.save))
```

Comment:
Note for review. THis is actually fully necessary (as well as the change in `test_operator` as calling an operator created with/without save with a different `TimeFunction` will lead to segafaults (calling a `save=..` with a `save=None` one) or fill the history if using a buffer for `save=nt`

Reply:
are you saying that if I have a ``save=None`` with say 3 buffer slots, and then I try to replace such a ``TimeFunction`` with another ``TimeFunction`` that this time has ``save=3``, we end up with a segfault?

**IF not** , then probably we still don't need ``save``. We could just use ``shape[0]``